instance/models/mixins/openedx_database.py

```diff
@@ -75,12 +117,58 @@ def mongo_database_names(self):
         """
         return [self.mongo_database_name, self.forum_database_name]
 
+    @property
+    def migrate_user(self):
+        """
+        Name of migration user
+        """
+        return self._get_mysql_user_name("migrate")
+
+    @property
+    def read_only_user(self):
+        """
+        Name of read_only user
+        """
+        return self._get_mysql_user_name("read_only")
+
+    @property
+    def admin_user(self):
+        """
+        Name of admin user
+        """
+        return self._get_mysql_user_name("admin")
+
+    @property
+    def global_users(self):
+        """
+        List of MySQL users with global privileges (i.e., privileges spanning multiple databases)
+        """
+        return self.migrate_user, self.read_only_user, self.admin_user
+
+    def _get_mysql_database_name(self, suffix):
+        """
+        Build unique name for MySQL database using suffix
+        """
+        return "{0}_{1}".format(self.mysql_database_name, suffix)
+
+    def _get_mysql_user_name(self, suffix):
+        """
+        Build unique name for MySQL user using suffix
+        """
+        return "{0}_{1}".format(self.mysql_user, suffix)
```

Comment:
We could also `assert` that the generated username is not longer than 16 characters.